src/server.c

```diff
@@ -3165,9 +3186,8 @@ void call(client *c, int flags) {
      * We leverage a static variable (prev_err_count) to retain
      * the counter across nested function calls and avoid logging
      * the same error twice. */
-    if ((server.stat_total_error_replies - prev_err_count) > 0) {
-        real_cmd->failed_calls++;
-    } else if (c->deferred_reply_errors) {
+    incrCommandFailedCalls(real_cmd, ERROR_COMMAND_FAILED);
```

Comment:
i'm having second thought about the name of this function.
it doesn't really (always) increment the error counter (we don't have an `if` when calling it).
so maybe a better name would be `IncrIfNeeded` or such.
and maybe it should have a return value, so the `if (c->deferred_reply_errors)` can return to being an `else if`?

maybe i need to sleep on it.